### PR TITLE
add users endpoint with controller model and tests

### PR DESCRIPTION
**Pull request by mkpi-pm**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -68,43 +68,67 @@ describe("/api", () => {
...
-  });
-  test("404: Returns the not found message when passed article id that doesn't exist", () => {
-    return request(app)
-      .get("/api/articles/99999999")
-      .expect(404)
-      .then(({ body }) => {
-        expect(body.msg).toBe("not found");
-      });
+  describe("PATCH /api/articles/:article_id ", () => {
+    test("201: Returns an object with updated votes property by the provided number", () => {
```
Comment (popatre) [inline __tests__/app.test.js]: This would be 200 - you've not created something, just amended.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -68,43 +68,67 @@ describe("/api", () => {
...
+          expect(body.msg).toBe("not found");
+        });
+    });
+    test("400: Responds with an error message when passed a bad user ID", () => {
+      return request(app)
+        .get("/api/articles/notAnID")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("bad request");
+        });
```
Comment (popatre) [inline __tests__/app.test.js:122]: Tests missing 

- [x] 400 inc_vote not an integer
- [x]  200/400 inc_vote key missing - either respond with an error or make no change. i.e they send an empty object

**Review by popatre — CHANGES_REQUESTED** [review]
Just needs a change and some additions to the patch one

File: __tests__/app.test.js (hunk 1 above)
Comment (mkpi-pm) [reply __tests__/app.test.js]: Yes, indeed. Thank you!

File: __tests__/app.test.js (hunk 2 above)
Comment (mkpi-pm) [reply __tests__/app.test.js:122]: After many glitches... done :)

Comment (Rosem10) [reply __tests__/app.test.js:122]: For the inc_vote key missing - we're looking for a test where you send just {} without the "inc_votes: " within it - and then handle the error accordingly, please!

**Review by Rosem10 — APPROVED** [review]

**Pull request merged**